### ISSUE TITLE
Change default device language not working

### ISSUE THREAD

**Issue by radinamatic** [learningequality/kolibri#5704]
### Observed behavior
This new device setting was [introduced recently](https://github.com/learningequality/kolibri/pull/5251), but when I set to [document it](https://github.com/learningequality/kolibri-docs/issues/100), I realized that it doesn't have the desired effects. 

### Expected behavior
Gherkin scenario previews the following behavior:
```
Scenario: Changing the default language in facility settings should result in that language being shown when the site is visited for the first time

GIVEN There is an admin user "Admin" and a learner user "Learner"
    AND "Admin" is at the Facility Settings Page
    AND The current default language is "English"
    AND "Learner" has never used Kolibri on their device before
WHEN "Admin" sets the default language to "French"
    AND "Admin" clicks *Save*
THEN "Admin" sees a notification that the settings have been successfully changed.
WHEN "Learner" accesses Kolibri for the first time
THEN The Sign In page (and the rest of the application) is localized in French 
```

From the above I presume that to test this it would be enough to try to access the server from the incognito session on the browser, correct?

### User-facing consequences
Admin not able to set the language for the learner's first sign in...?

### Errors and logs
…

### Steps to reproduce
1. Install Kolibri (I used a VM, and went through the setup wizard in Spanish)
2. Go to Device > Settings and select a different language as default. In my case `Español` was preselected, probably because it was used during the setup wizard, but I changed it to Fulfulde and saved.
3. Access the sign-in page from an incognito browser or another device. The result in my case was always English.

`api/device/devicelanguage/` seems correctly set, but that does not reflect in the other browsers/ devices who access the server IP for the first time.

![Selection_126](https://user-images.githubusercontent.com/1457929/60106761-0baf3180-9766-11e9-9abc-b4f005852f85.png)

### Context
 * Kolibri version: 0.12.5
 * Operating system: Ubuntu 16.04
 * Browser: Chrome & FF

cc @jonboiser

**Comment by jonboiser** [comment]
Looked into the code in #5251, and I see that the this `DEVICE_LANGUAGE_CACHE_KEY` value gets cached for about 10 minutes. 

https://github.com/learningequality/kolibri/blob/a618e8740909f30de65e2e0d8aa484ce07a189f0/kolibri/core/device/translation.py#L27-L40

In #5251, I notice that I don't clear out the cache when changing languages. But you should be able to skip the 10 minute wait if you restart the server.

**Comment by rtibbles** [comment]
Hrm - I wonder if we should move this logic onto the DeviceSetting model, and consolidate any updates to the language setting there? Happy to take this on.

**Comment by jonboiser** [comment]
That sounds like a good refactor to me :+1:

**Comment by indirectlylit** [comment]
see related issue: https://github.com/learningequality/kolibri/issues/5727

**Comment by rtibbles** [comment]
Ah, this is not a cache clearing problem - the cache does get cleared in the `DeviceSetting` save method.

This will be fixed by #5746

**Comment by rtibbles** [comment]
Fixed in #5746
